src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Deployments/StopAzureDeploymentCmdlet.cs

```diff
@@ -0,0 +1,74 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Components;
+using Microsoft.Azure.Commands.ResourceManager.Cmdlets.SdkModels;
+using ProjectResources = Microsoft.Azure.Commands.ResourceManager.Cmdlets.Properties.Resources;
+
+namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation
+{
+    /// <summary>
+    /// Cancel a running deployment.
+    /// </summary>
+    [Cmdlet(VerbsLifecycle.Stop, "AzureRmDeployment", SupportsShouldProcess = true,
+        DefaultParameterSetName = StopAzureDeploymentCmdlet.DeploymentNameParameterSet), OutputType(typeof(bool))]
+    public class StopAzureDeploymentCmdlet : ResourceManagerCmdletBase
+    {
+        /// <summary>
+        /// The deployment Id parameter set.
+        /// </summary>
+        internal const string DeploymentIdParameterSet = "StopByDeploymentId";
+
+        /// <summary>
+        /// The deployment name parameter set.
+        /// </summary>
+        internal const string DeploymentNameParameterSet = "StopByDeploymentName";
+
+        /// <summary>
+        /// The input object parameter set.
+        /// </summary>
+        internal const string InputObjectParameterSet = "StopByInputObject";
+
+        [Alias("DeploymentName")]
+        [Parameter(Position = 0, ParameterSetName = StopAzureDeploymentCmdlet.DeploymentNameParameterSet,
+            Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The name of the deployment.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Alias("DeploymentId", "ResourceId")]
+        [Parameter(ParameterSetName = StopAzureDeploymentCmdlet.DeploymentIdParameterSet, Mandatory = true,
+            HelpMessage = "The fully qualified resource Id of the deployment. example: /subscriptions/{subId}/providers/Microsoft.Resources/deployments/{deploymentName}")]
+        [ValidateNotNullOrEmpty]
+        public string Id { get; set; }
+
+        [Parameter(ParameterSetName = StopAzureDeploymentCmdlet.InputObjectParameterSet, Mandatory = true,
+            ValueFromPipeline = true, HelpMessage = "The deployment object.")]
+        public PSDeployment InputObject { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            var deploymentName = !string.IsNullOrEmpty(this.Name)
+                ? this.Name
+                : !string.IsNullOrEmpty(this.Id) ? ResourceIdUtility.GetResourceName(this.Id) : this.InputObject.DeploymentName;
+
+            ConfirmAction(
+                ProjectResources.CancelDeploymentMessage,
+                this.Name,
+                () => ResourceManagerSdkClient.CancelDeploymentAtSubscriptionScope(deploymentName));
+
+            WriteObject(true);
```

Comment:
@Tiano2017 same comment about implementing the `-PassThru` parameter

Reply:
same. was following what we do for Stop-AzureRmResourceGroupDeployment.

Reply:
No, we need to make this meet the design guidelines now.